dashboard/src/pages/hardwareDetails/HardwareDetails.tsx

```diff
@@ -14,19 +16,55 @@ import {
 import { Skeleton } from '@/components/Skeleton';
 import { useHardwareDetails } from '@/api/hardwareDetails';
 
+import type { Trees } from '@/types/hardware/hardwareDetails';
+
 import { HardwareHeader } from './HardwareDetailsHeaderTable';
 import HardwareDetailsTabs from './Tabs/HardwareDetailsTabs';
 
+const sanitizeTreeItems = (
+  treeItems: Record<string, string>[] | undefined,
+): Trees[] => {
+  if (!treeItems) return [];
+  return treeItems.map((tree, index) => ({
+    treeName: tree['tree_name'] ?? '-',
+    gitRepositoryBranch: tree['git_repository_branch'] ?? '-',
+    gitCommitName: tree['git_commit_name'] ?? '-',
+    gitCommitHash: tree['git_commit_hash'] ?? '-',
+    gitRepositoryUrl: tree['git_repository_url'] ?? '-',
+    sortID: tree['sort_id'] ?? index.toString(),
+  }));
+};
+
 function HardwareDetails(): JSX.Element {
+  const { intervalInDays, origin } = useSearch({ from: '/hardware' });
   const searchParams = useSearch({ from: '/hardware/$hardwareId' });
+  const { treeFilter: treeIndexes } = searchParams;
+
   const { hardwareId } = useParams({ from: '/hardware/$hardwareId' });
-  const { intervalInDays, origin } = useSearch({ from: '/hardware' });
+
+  const navigate = useNavigate({ from: '/hardware/$hardwareId' });
+
+  const updateTreeFilters = useCallback(
+    (selectedIndexes?: number[]) => {
+      navigate({
+        search: previousSearch => ({
+          ...previousSearch,
+          treeFilter: selectedIndexes,
+        }),
+      });
+    },
+    [navigate],
+  );
+
   const { data, isLoading } = useHardwareDetails(
     hardwareId,
     intervalInDays,
     origin,
+    treeIndexes ?? [],
```

Comment:
When we enter in `HardwareDetails` for the first time, the `treeIndexes` is `undefined` and a empty list of trees ids is passed as params. But initially, all trees are selected. We should pass all the trees ids in this case.